### PR TITLE
Add crafting to the farmer

### PR DESCRIPTION
**Pull request by Mekle001**
# Changes proposed in this pull request:
- Make the farmer a public crafter
- Add carved pumpkin recipe to farmer
- Allow the farmer to make Hay Bales and seeds
- Add smooth sandstone variants to stone smelter
- Add prismarine shards/crystals to stone mason
- Add Sea Lantern to Mechanic

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/farmer/EntityAIWorkFarmer.java [hunk 1]
```diff
@@ -131,6 +135,46 @@ public EntityAIWorkFarmer(@NotNull final JobFarmer job)
...
+        if (job.getTaskQueue().isEmpty())
+        {
+            return PREPARING;
+        }
+
+        if (job.getCurrentTask() == null)
+        {
+            return PREPARING;
+        }
+
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/farmer/EntityAIWorkFarmer.java:168]: Yu might need the same check here I added to the plantation.

Comment (Mekle001) [reply src/main/java/com/minecolonies/coremod/entity/ai/citizen/farmer/EntityAIWorkFarmer.java:168]: I believe your change was to the AbstractEntityAICrafting, which is called at the beginning here, so should cover the farmer too.

Comment (marchermans) [reply src/main/java/com/minecolonies/coremod/entity/ai/citizen/farmer/EntityAIWorkFarmer.java:168]: Verify that this is indeed the case.

**Review by marchermans — APPROVED** [review]

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/farmer/EntityAIWorkFarmer.java (hunk 1 above)
Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/entity/ai/citizen/farmer/EntityAIWorkFarmer.java:168]: Yes, I double checked

**Pull request merged**